### ISSUE TITLE
Gateway v2 validation rejects nullable query variable used as input to non-null field of input object with a default value

### ISSUE THREAD

**Issue by cysp** [apollographql/federation#2174]
## Summary

Hi, we've got a schema with a field that takes an input object where a field of that input object is non-nullable and has a default value. We make a query for that field with a _nullable_ query variable, so that we don't need to repeat the default value at the client.
This was accepted by Apollo Gateway v0 and is still accepted by Apollo Server v3 and v4, but is rejected by a validation implemented in Apollo Gateway v2. TBH I don't know if this is supposed to be valid according to the graphql spec or if it has just slipped past validations thus far.

## Reduced testcase

Runnable example in gist: https://gist.github.com/cysp/7a9d8b5cfe808315b24ece291693afd8

### Schema:
```graphql
input Input {
  a: Int! = 1234
}

type Query {
  field(input: Input!): String
}
```

### Operation:
```graphql
query Foo($a: Int) {
  field(input: { a: $a })
}
```

## Expected result

The query is accepted and `input.a` in the field resolver takes on the default value of the input field, 1234

## Actual result

```json
{
  "errors": [
    {
      "message": "Invalid value {a: $a} for argument \"Query.field(input:)\" of type Input!",
      "extensions": […]
    }
  ]
}
```

**Comment by cysp** [comment]
Hm, after raising this, I see that it could be related to https://github.com/apollographql/federation/issues/1959

**Comment by pcmanus** [comment]
> Hm, after raising this, I see that it could be related to https://github.com/apollographql/federation/issues/1959
 
No worries, and in fact I think this is genuinely a different issue.

And somewhat regardless of the details of the graphQL spec, I agreed that this something we should change (back): that use case seems perfectly reasonable and useful. Probably a fairly easy fix too, I'll have a look.

**Comment by cysp** [comment]
@pcmanus I believe this was released in v2.1.4 but I'm still seeing this operation rejected by `@apollo/gateway`
I've updated the gist linked above with the newly released package versions to show the current behaviour.

The problem appears to be that `field.defaultValue` is undefined inside the validation function:
https://github.com/pcmanus/federation/blob/e0b7bf14d2fe83155adaccda801bcfce82fe3d5d/internals-js/src/values.ts#L528

It appears that we lose default values when copying input object field definitions from the super graph to the subgraph, but I can't say that this is the _only_ place that this occurs. The following branch passes my local testing:
https://github.com/apollographql/federation/compare/main...cysp:apollographql-federation:fix/extract-subgraph-input-field-default-value

**Comment by pcmanus** [comment]
Thanks @cysp, you're absolutely right, I missed that part.
